SHA3.cs

```diff
@@ -0,0 +1,191 @@
+namespace System.Security.Cryptography;
+
+public partial struct HashAlgorithmName {
+    public static HashAlgorithmName SHA3_256 { get; }
+    public static HashAlgorithmName SHA3_384 { get; }
+    public static HashAlgorithmName SHA3_512 { get; }
+}
+
+public sealed partial class RSAEncryptionPadding {
+    public static RSAEncryptionPadding OaepSHA3_256 { get; }
+    public static RSAEncryptionPadding OaepSHA3_384 { get; }
+    public static RSAEncryptionPadding OaepSHA3_512 { get; }
+}
+
+public abstract partial class SHA3_256 : HashAlgorithm {
```

Comment:
The SHA-3 follow our existing SHA-2 design, with two exceptions:

1. The `IsSupported` property. This property will tell developers if the algorithm is available (and functional) on their platform.
2. Deprecated APIs will not be brought over.